common/src/main/java/feast/common/logging/AuditLogger.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright 2018-2020 The Feast Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package feast.common.logging;
+
+import com.google.protobuf.Message;
+import feast.common.logging.config.LoggingProperties;
+import feast.common.logging.config.LoggingProperties.AuditLogProperties;
+import feast.common.logging.entry.ActionAuditLogEntry;
+import feast.common.logging.entry.AuditLogEntry;
+import feast.common.logging.entry.AuditLogEntry.FeastInstance;
+import feast.common.logging.entry.AuditLogEntryKind;
+import feast.common.logging.entry.LogResource;
+import feast.common.logging.entry.LogResource.ResourceType;
+import feast.common.logging.entry.MessageAuditLogEntry;
+import feast.common.logging.entry.TransitionAuditLogEntry;
+import lombok.extern.slf4j.Slf4j;
+import org.slf4j.Marker;
+import org.slf4j.MarkerFactory;
+import org.slf4j.event.Level;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+public class AuditLogger {
+  private static final Marker AUDIT_MARKER = MarkerFactory.getMarker("AUDIT_MARKER");
+  private static FeastInstance instance;
+  private static AuditLogProperties properties;
+
+  @Autowired
+  public AuditLogger(FeastInstance instance, LoggingProperties properties) {
+    // Spring runs this constructor when creating the AuditLogger bean,
+    // which allows us to populate the AuditLogger class with dependencies.
+    // This allows us to use the dependencies in the AuditLogger's static methods
+    AuditLogger.instance = instance;
+    AuditLogger.properties = properties.getAudit();
+  }
+
+  /**
+   * Log the handling of a Protobuf message by a service call.
+   *
+   * @param level describing the severity of the log.
+   * @param method the name of the method handling the service call
+   * @param request protobuf message in the service call.
+   * @param response protobuf message in the servicef call.
+   */
+  public static void logMessage(
+      Level level, String method, Message request, Message response, String identity) {
+    log(
+        level,
+        MessageAuditLogEntry.newBuilder()
```

Comment:
If we don't allow direct access to logging AuditLogEntry classes (or its children), then is there an advantage to using the builder pattern?

Reply:
The builder pattern is useful for `MessageAuditLogEntry` in specific as it can replace this class. https://github.com/feast-dev/feast/blob/d03a46523e4894806f407a4db290c4d27511cfec/common/src/main/java/feast/common/interceptors/GrpcMessageInterceptor.java#L41.

We can reintroduce `of...()` construction methods to use where verbosity is more of a concern.

Reply:
Reintroduced of() methods for `TransitionAuditLogEntry` and `ActionAuditLogEntry`.